src/pytorch_lightning/callbacks/sanity_serving.py

```diff
@@ -0,0 +1,92 @@
+import time
+from modulefinder import Module
+from multiprocessing import Process
+from typing import Any, Callable, Dict, Optional, Tuple
+
+import requests
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.utilities.imports import requires
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+
+class ServableModule(Module):
+    def configure_payload(self) -> Dict[str, Any]:
+        ...
+
+    def configure_inputs_outputs(self) -> Tuple[Dict[str, Callable], Dict[str, Callable]]:
+        ...
+
+    def serve_step(self, *args, **kwargs):
+        ...
+
+
+def _start_server(servable_model: ServableModule, host: str, port: int, method: bool) -> None:
+    """This method starts a simple FastAPI server with a predict and ping endpoint."""
+    from fastapi import Body, FastAPI
+    from uvicorn import run
+
+    app = FastAPI()
+
+    inputs, outputs = servable_model.configure_inputs_outputs()
+
+    @app.get("/ping")
+    def ping() -> bool:
+        return True
+
+    @app.post("/predict")
+    async def predict(payload: dict = Body(...)) -> Dict[str, Any]:
+        body = payload["body"]
+
+        for key, deserializer in inputs.items():
+            body[key] = deserializer(body[key])
+
+        with torch.no_grad():
+            output = servable_model.serve_step(**body)
+
+        if not isinstance(output, dict):
+            raise Exception(f"Please, return your outputs as a dictionary. Found {output}")
+
+        for key, serializer in outputs.items():
+            output[key] = serializer(output[key])
+
+        return output
+
+    run(app, host=host, port=port, log_level="error")
+
+
+class SanityServing(Callback):
+
+    """This callback enables to validate a model is servable following the ServableModule API."""
+
+    @requires("fastapi")
+    @requires("uvicorn")
```

Comment:
Can we avoid this new decorator and instead use the `_RequirementAvailable` class we've recently started using for these? We are trying to standardize all these utilities into one: #12786 

In the past, William has been very against decorators.

example:
```python
FASTAPI_AVAILABLE = _RequirementAvailable("fastapi")

if not FASTAPI_AVAILABLE:
    ...
```

Reply:
@williamFalcon was against decorator for the users, not internal usage. I don't like `_RequirementAvailable`, I would prefer requires if we need to uniformize something.